gene2phenotype_project/gene2phenotype_app/views/locus_genotype_disease.py

```
@@ -998,17 +1021,34 @@ def post(self, request, stable_id):
         if not user_serializer.check_panel_permission(lgd_panels):
             return Response({"message": f"No permission to edit {stable_id}"}, status=status.HTTP_403_FORBIDDEN)
 
-        comment = request.data.get("comment", None)
-        is_public = request.data.get("is_public", None)
+        # LGDCommentListSerializer accepts a list of comments
+        serializer_list = LGDCommentListSerializer(data=request.data)
+
+        if serializer_list.is_valid():
+            lgd_comments_data = serializer_list.validated_data.get("comments")
+
+            if(not lgd_comments_data):
+                response = Response(
+                    {"message": "Empty comment. Please provide valid data."},
+                     status=status.HTTP_400_BAD_REQUEST
+                )
+
+            # Add each comment from the input list
+            for comment in lgd_comments_data:
+                serializer_class = LGDCommentSerializer(
+                    data=comment,
+                    context={"lgd": lgd, "user": user_obj}
+                )
+
+                if serializer_class.is_valid():
+                    serializer_class.save()
+                else:
+                    response = Response({"errors": serializer_class.errors}, status=status.HTTP_400_BAD_REQUEST)
```

If we are saving 2 comments, if 1st comment failed to save, the response will be set to error response but if the 2nd comment is successful, the response will be overidden to success response. It should return the error response on failure of 1st comment right?

---

If the first comment fails because it's already associated with the record, the endpoint returns BAD REQUEST with the following message:
`"Comment is already associated with '<stable_id>'"`

---

Actually any serializer_class.errors will be returned with 400 BAD REQUEST